react-app/src/app/screens/Game/index.js

```diff
@@ -1,18 +1,71 @@
 import React, { Component } from 'react';
 
 import Board from './components/Board';
+import { calculateWinner } from './utils';
 import style from './styles.scss';
 
 class Game extends Component {
+  state = {
+    history: [
+      {
+        squares: Array(9).fill(null)
+      }
+    ],
+    stepNumber: 0,
+    xIsNext: true
+  };
+
+  handleClick = i => {
+    const history = this.state.history.slice(0, this.state.stepNumber + 1);
+    const current = history[history.length - 1];
+    const squares = [...current.squares];
+    const hasWinner = calculateWinner(squares);
+    const squareIsFilled = squares[i];
+    if (hasWinner || squareIsFilled) return;
+    squares[i] = this.state.xIsNext ? 'X' : 'O';
+    this.setState({
+      history: history.concat([
+        {
+          squares
+        }
+      ]),
+      xIsNext: !this.state.xIsNext,
+      stepNumber: history.length
+    });
+  };
+
+  jumpTo(step) {
+    this.setState({
+      stepNumber: step,
+      xIsNext: step % 2 === 0
+    });
+  }
+
   render() {
+    const history = this.state.history;
+    const current = history[this.state.stepNumber];
+    const winner = calculateWinner(current.squares);
+
+    const moves = history.map((step, move) => {
+      const text = move ? `Go to move #${move}` : 'Go to game start';
+      return (
+        <li key={move}>
+          <button onClick={() => this.jumpTo(move)}>{text}</button>
+        </li>
+      );
+    });
+
+    let status;
+    if (winner) status = `Winner: ${winner}`;
+    else status = `Next player: ${this.state.xIsNext ? 'X' : 'O'}`;
     return (
       <div className={style.game}>
         <div>
-          <Board />
+          <Board squares={current.squares} onClick={this.handleClick} />
         </div>
         <div className={style.gameInfo}>
-          <div>{/* status */}</div>
-          <ol>{/* TODO */}</ol>
+          <div>{status}</div>
+          <ol>{moves}</ol>
         </div>
       </div>
```

Comment:
This render is having lots of logic, how about moving it to other functions. Furthermore, this class is having a lot of logic and a view. You should move the layout into a layout.js. And there put the layout of the component, which in this case will be this
```
<div>
....
</div>
```